### PR TITLE
Added support of the NSSL CCPP suite - WoFS_v0 and WE2E tests

### PR DESCRIPTION
**Pull request by ywangwof**
## DESCRIPTION OF CHANGES: 
Added support for the NSSL CCPP suite, WoFS_v0 in the release branch.

## TESTS CONDUCTED: 
Tested on Odin using Intel 2020 and Jet with the predefined grid RRFS_CONUS_3km and RRFS_SUBCONUS_3km.

## DEPENDENCIES:
It depends on the following PRs
- ufs-community/ufs-weather-model/pull/1220
- NOAA-EMC/fv3atm/pull/535

## MODIFIED AND NEW FILES:
scripts/exregional_make_ics.sh
scripts/exregional_make_lbcs.sh
ush/templates/FV3.input.yml
ush/templates/diag_table.FV3_WoFS_v0
ush/templates/field_table.FV3_WoFS_v0
ush/valid_param_vals.sh
tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0.sh
tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0.sh

## ISSUE (optional): 

- This addition will require an upgrade of the ufs-weather-model repository in the App level (see above).

- The runtime files for this workflow contains new sections for HAILCAST output. The code update of GFDL_atmos_cubed_sphere has already been merged with PR [#164](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/164). 

## CONTRIBUTORS (optional): 
Ted Mansell (@MicroTed) provided the initial runtime files for this workflow.

### CODE REVIEW

**Review by RatkoVasic-NOAA — APPROVED** [review]

**Comment by JeffBeck-NOAA** [comment]
Hi @daniel-hildebrandt, this PR needs to specifically go into the release/public-v2 branch of regional_workflow.  The WoFS SDF won't be going into the develop branch of ufs-weather-model for the moment, and will only go into the release branch of that repo.  Therefore, we can only merge this PR to the release/public-v2 branch of regional_workflow, once it is recreated.  I'll go ahead and reopen it with this comment, but please let me know if there are concerns!

**Comment by daniel-hildebrandt** [comment]
> 

@JeffBeck-NOAA Thank you for the info! I recreated the public-v2 release branch

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#
```
Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0.sh]: This test is on the relatively large `RRFS_SUBCONUS_3km` grid (840x600).  Is this the grid you want, or did you want the much smaller `SUBCONUS_Ind_3km` grid (which is 200x200)?  For rapid testing purposes, the latter would be better.

Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0.sh:1]: This test is on the relatively large `RRFS_SUBCONUS_3km` grid (840x600).  Is this the grid you want, or did you want the much smaller `SUBCONUS_Ind_3km` grid (which is 200x200)?

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#
```
Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0.sh]: As above regarding grids:  `RRFS_SUBCONUS_3km` vs. `SUBCONUS_Ind_3km`

Comment (gsketefian) [inline tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0.sh:1]: As above:  `RRFS_SUBCONUS_3km` vs. `SUBCONUS_Ind_3km`

**Comment by gsketefian** [comment]
@ywangwof There are some new files introduced by this PR that look like came from other PRs that haven't yet been merged into the release branch:
```
.github/CODEOWNERS
.github/workflows/python_unittests.yaml
tests/WE2E/create_WE2E_resource_summary.py
```
You might want to remove those (or wait until their PRs are merged into the release branch).

**Comment by gsketefian** [comment]
@ywangwof There are some new files introduced by this PR that look like came from other PRs that haven't yet been merged into the release branch:
```
.github/CODEOWNERS
.github/workflows/python_unittests.yaml
tests/WE2E/create_WE2E_resource_summary.py
```
You might want to remove those (or wait until their PRs are merged into the release branch).

**Comment by ywangwof** [comment]
> 
Those files are in branch **release/public-v2**
```
.github/CODEOWNERS added on 2022-05-12 (#757)
.github/workflows/python_unittests.yaml added on 2022-05-16 (#747)
tests/WE2E/create_WE2E_resource_summary.py added on 2022-05-16 (#769)
```

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0.sh (hunk 2 above)
Comment (ywangwof) [reply tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0.sh]: The two WE2E test files are change to use the smaller domain, `SUBCONUS_Ind_3km` and tested. Thanks.

Comment (gsketefian) [reply tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0.sh]: Ok, thanks.

File: tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0.sh (hunk 1 above)
Comment (gsketefian) [reply tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0.sh]: Ok, thanks.

**Comment by gsketefian** [comment]
> > 
> 
> Those files are in branch **release/public-v2**
> 
> ```
> .github/CODEOWNERS added on 2022-05-12 (#757)
> .github/workflows/python_unittests.yaml added on 2022-05-16 (#747)
> tests/WE2E/create_WE2E_resource_summary.py added on 2022-05-16 (#769)
> ```

Ok, I guess `.github/CODEOWNERS` already existed and is showing up in your PR as modified, but the other two are showing up as new when I view the list of files.  It's strange.  I'll leave it up to @mark-a-potts to merge this if this inconsistency is ok with him.

**Review by gsketefian — APPROVED** [review]

**Pull request merged**